### PR TITLE
creating an basic app with express and a psql database #9 #6

### LINKED ISSUE #6: Creating a build file for psql db, and setting a connection file for the db
Creating a build file for psql db, and setting a connection file for the db.
so we could create the db locally and use the data for development.

### LINKED ISSUE #9: Setting up a server using express
creating the basic structure for the server using express.
it will include all that necessary to run a basic express server.

### PR DESCRIPTION
**Pull request by idantb101**
- [ ] Creating a base app that includes express and PSQL server build script. 

- [ ] closes #9 
- [ ] closes #6

### CODE REVIEW

**Comment by harrygfox** [comment]
@idantb101 Is this one ready to be reviewed?

**Comment by NickP123** [comment]
Happy with the code. So glad we got out of PSQL hell... : 0 )

**Comment by NickP123** [comment]
Yes ready for review Harry

File: package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  "scripts": {
+    "start": "nodemon app.js"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "body-parser": "^1.18.2",
+    "env2": "^2.2.0",
+    "express": "^4.16.2",
+    "nodemon": "^1.14.11",
```
Comment (Karyum) [inline package.json]: this supposed to be a dev dep and not in dep

File: src/db/db_build.sql [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    diseaseName VARCHAR(100) NOT NULL
+);
+
+INSERT INTO diseases (diseaseName)
+VALUES ('cancer');
+
+CREATE TABLE tags (
+    id SERIAL PRIMARY KEY,
+    tagName VARCHAR(50) NOT NULL,
+    diseaseId INTEGER REFERENCES diseases(id)
```
Comment (Karyum) [inline src/db/db_build.sql]: can you give me an explanation on what ` INTEGER REFERENCES diseases(id)` would do ?

Comment (Karyum) [reply src/db/db_build.sql]: and whats the difference between it and `FOREIGN KEY`?

File: src/db/db_build.sql [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+);
+
+INSERT INTO tags (tagName, diseaseId)
+VALUES ('cancer', 1);
+
+CREATE TABLE users (
+    id SERIAL PRIMARY KEY,
+    fullName VARCHAR(40) NOT NULL,
+    email VARCHAR(50) NOT NULL,
+    password VARCHAR NOT NULL,
```
Comment (Karyum) [inline src/db/db_build.sql]: you have just `VARCHAR` here without specifying how long it should or how many number of characters should it include

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,62 @@
+nconfig.env
```
Comment (claireinez) [inline .gitignore]: First two lines have typos but they're not needed anyway as they're duplicated below

File: src/db/db_build.sql (hunk 2 above)
Comment (idantb101) [reply src/db/db_build.sql]: I don't have a better answer than what Finn said and I trust him.
I can change it if it is important to you ^-^

File: src/db/db_build.sql (hunk 3 above)
Comment (idantb101) [reply src/db/db_build.sql]: because passwords are encrypted how long should we allow them to be?

File: src/db/db_build.sql (hunk 2 above)
Comment (Karyum) [reply src/db/db_build.sql]: > I don't have a better answer than what Finn said and I trust him.

i don't :laughing: 
but if it behaves like foreign key and **it works** then its fine by me

File: src/db/db_build.sql (hunk 3 above)
Comment (Karyum) [reply src/db/db_build.sql]: i guess my main question here does it work without specifying the limit?

File: src/db/db_build.sql (hunk 2 above)
Comment (idantb101) [reply src/db/db_build.sql]: :+1:

File: src/db/db_build.sql (hunk 3 above)
Comment (idantb101) [reply src/db/db_build.sql]: yes then it uses the default limit

**Comment by idantb101** [comment]
Thanks :)

**Pull request merged**